test/service/test_manifest_async.py

```diff
@@ -310,37 +311,99 @@ def test(self, mock_uuid, mock_helper):
+        step_function_helper.describe_execution.return_value = {
+            'status': 'RUNNING'
+        }
+        url = self.base_url.set(path='/fetch/manifest/files')
+        location = set()
+        status = set()
+        for i in range(3):
+            response = requests.get(str(url)).json()
+            location.add(response['Location'])
+            status.add(response['Status'])
+        url.args.add('token',
+                     Token(execution_id=self.object_key + '_0',
+                           request_index=1,
+                           wait_time=1).encode())
+        self.assertEqual(str(url), one(location))
+        self.assertEqual(301, one(status))
+
+    @patch_step_function_helper
+    def test_idempotent_manifest_request_retry(self, step_function_helper):
+        """
+        Multiple request for the same manifest shouldn't trigger multiple
+        step function executions.
+        """
+        client_error = self._client_error('ExecutionAlreadyExists')
+        step_function_helper.start_execution.side_effect = [client_error, 0]
+        step_function_helper.describe_execution.return_value = {
+            'status': 'SUCCEEDED'
+        }
+        url = self.base_url.set(path='/fetch/manifest/files')
+        response = requests.get(str(url)).json()
+
+        partition = ManifestPartition.first()
+        step_function_helper.start_execution.assert_has_calls([
+            mock.call(config.state_machine_name('manifest'),
+                      f'{self.object_key}_{i}',
+                      execution_input=dict(format_='compact',
+                                           catalog='test',
+                                           filters={},
+                                           object_key=self.object_key,
+                                           partition=partition.to_json()))
+            for i in range(2)
+        ])
+
+        url.args.add('token',
+                     Token(execution_id=f'{self.object_key}_1',
+                           request_index=0,
+                           wait_time=1).encode())
+        self.assertEqual(str(url), response['Location'])
+        self.assertEqual(301, response['Status'])
+
     @patch_step_function_helper
     def test_execution_not_found(self, step_function_helper):
         """
         Manifest status check should raise a BadRequestError (400 status code)
         if execution cannot be found.
         """
-        step_function_helper.describe_execution.side_effect = ClientError({
-            'Error': {
-                'Code': 'ExecutionDoesNotExist'
-            }
-        }, '')
-        url = self.base_url.set(path='/fetch/manifest/files', args=self.params)
+        client_error = self._client_error('ExecutionDoesNotExist')
+        step_function_helper.describe_execution.side_effect = client_error
+        url = self.base_url.set(
+            path='/fetch/manifest/files', args=self.params)
```

Comment:
Fits on one line.
```suggestion
        url = self.base_url.set(path='/fetch/manifest/files', args=self.params)
```
If it didn't, the proper way to wrap would be:
```suggestion
        url = self.base_url.set(path='/fetch/manifest/files', 
                                args=self.params)
```